apps/stripe/src/app/api/webhooks/stripe/stripe-object-handlers/stripe-payment-intent-handler.ts

```diff
@@ -113,18 +127,60 @@ export class StripePaymentIntentHandler {
     return ok(recordedTransactionResult.value);
   }
 
+  private async getPaymentMethodDetails(
+    saleorSchemaVersion: SaleorSchemaVersion,
+    stripePaymentIntentsApi: IStripePaymentIntentsApi,
+    paymentIntentId: StripePaymentIntentId,
+  ): Promise<SaleorPaymentMethodDetails | null> {
+    const validator = new SaleorVersionCompatibilityValidator(PAYMENT_METHOD_DETAILS_MIN_VERSION);
+
+    if (validator.isSaleorCompatible(saleorSchemaVersion)) {
```

Comment:
IMO it's better to check on parent method if its compatible, so this method complexity is lower - it will only resolve data